### PR TITLE
fix: amplify app version regex

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

Fix amplify app regex to capture `<major>.<minor>.<fix>` prefix correctly.

Previous version was incorrectly processing tag/rc versions.
image

#### Description of how you validated changes

CB:

https://us-east-1.console.aws.amazon.com/codesuite/codebuild/671107461633/projects/AmplifyCLI-E2E-Testing/batch/AmplifyCLI-E2E-Testing%3Af6bdb9ac-5a93-4898-86d1-a01f1907a4b5?region=us-east-1

image

CCI:
https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli?branch=run-e2e%2Fsobkamil%2Ffix-modelgen-test

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.base.yml [hunk 1]
```diff
@@ -1469,7 +1469,9 @@ commands:
                   source $BASH_ENV
             - run:
                 name: Confirm Amplify CLI is installed and available in PATH
-                command: amplify version
+                command: |
+                  which amplify
```
Comment (sobolk) [inline .circleci/config.base.yml:1473]: Added this to confirm that `amplify` resolves right path. I find it useful for troubleshooting, hence leaving it.

**Review by awsluja — DISMISSED** [review]
looks good; if with a good e2e run

**Review by edwardfoyle — DISMISSED** [review]

File: packages/amplify-app/src/index.js [hunk 2]
```diff
@@ -112,7 +112,7 @@ async function installAmplifyCLI() {
 async function amplifyCLIVersionCheck() {
   try {
     const amplifyCLIVersionRaw = await callAmplify(['-v'], { inheritIO: false });
-    const amplifyCLIVersionMatch = amplifyCLIVersionRaw.match(/\d+\.\d+\.\d+(-[a-z]+\.[0-9]+)?/g);
+    const amplifyCLIVersionMatch = amplifyCLIVersionRaw.match(/^\d+\.\d+\.\d/g);
```
Comment (jhockett) [inline packages/amplify-app/src/index.js:115]: The last `+` looks important to me unless I'm missing something

```suggestion
    const amplifyCLIVersionMatch = amplifyCLIVersionRaw.match(/^\d+\.\d+\.\d+/g);
```

**Review by jhockett — CHANGES_REQUESTED** [review]

File: packages/amplify-app/src/index.js (hunk 2 above)
Comment (sobolk) [reply packages/amplify-app/src/index.js]: it is, good catch.

**Review by jhockett — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**